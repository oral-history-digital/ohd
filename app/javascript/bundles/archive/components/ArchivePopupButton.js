import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';

import { openArchivePopup } from '../actions/archivePopupActionCreators';

export default function ArchivePopupButton({title, children, buttonFaKey}) {

    const dispatch = useDispatch();
    const open = useCallback(
        () => dispatch(openArchivePopup({title: title, content: children})),
        [dispatch]
    )

    return (
        <span
            className='flyout-sub-tabs-content-ico-link'
            title={title}
            onClick={open}
        >
            <i className={`fa fa-${buttonFaKey}`}></i>
            {title}
        </span>
    )
}

ArchivePopupButton.propTypes = {
    title: PropTypes.string,
    buttonFaKey: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.string,
    ]),
};
