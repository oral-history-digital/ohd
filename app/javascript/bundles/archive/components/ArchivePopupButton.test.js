import React from 'react';
import renderer from 'react-test-renderer';

import ArchivePopupButton from './ArchivePopupButton';

const mockedState = {
    data: {
        accounts: {
            current: {
                admin: true
            }
        }
    }
};

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn().mockImplementation(() => mockedState)
  }));

it('renders correctly', () => {
    const tree = renderer
        .create(
            <ArchivePopupButton
                title='Bearbeiten'
                buttonFaKey='pencil'
            >
                {'Klaus'}
            </ArchivePopupButton>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

