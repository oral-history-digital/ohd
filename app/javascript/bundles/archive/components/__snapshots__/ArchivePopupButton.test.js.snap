// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="flyout-sub-tabs-content-ico-link"
  onClick={[Function]}
  title="Bearbeiten"
>
  <i
    className="fa fa-pencil"
  />
  Bearbeiten
</span>
`;
